test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpMetricsExporterTests.cs

```diff
@@ -234,7 +234,7 @@ public void TestGaugeToOtlpMetric(string name, string description, string unit,
         using var meter = new Meter(Utils.GetCurrentMethodName());
         using var provider = Sdk.CreateMeterProviderBuilder()
             .AddMeter(meter.Name)
-            .SetExemplarFilter(enableExemplars ? new AlwaysOnExemplarFilter() : new AlwaysOffExemplarFilter())
+            .SetExemplarFilter(enableExemplars ? ExemplarFilterType.AlwaysOn : ExemplarFilterType.AlwaysOff)
```

Comment:
It’s not a part of this PR. I observed that we test AlwaysOn or AlwaysOff on `SetExemplarFilter`. Is there a reason why TraceBased is not checked anywhere?

Reply:
I added a case here: https://github.com/open-telemetry/opentelemetry-dotnet/pull/5404/commits/6963b0c21caa179335165ef7de3d8bf86ac681a5

I don't think there was a reason for that other than the test coverage for exemplars has been lacking in general.